README.md

```diff
@@ -122,24 +122,101 @@ For more information regarding the configuration of Open Liberty's logging capab
+        && mvn -f /opt/ol/wlp/usr/shared/resources/infinispan/pom.xml dependency:copy-dependencies -DoutputDirectory=/opt/ol/wlp/usr/shared/resources/infinispan \
+        && apt-get remove -y maven \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/pom.xml \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/jboss-transaction-api*.jar \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/reactive-streams-*.jar \
+        && rm -f /opt/ol/wlp/usr/shared/resources/infinispan/rxjava-*.jar \
+        && rm -rf ~/.m2 \
+        && chown -R 1001:0 /opt/ol/wlp/usr/shared/resources/infinispan \
+        && chmod -R g+rw /opt/ol/wlp/usr/shared/resources/infinispan
+    USER 1001
+    
+    FROM open-liberty AS open-liberty-infinispan
+    USER root
+    
+    # Copy Infinispan client jars to Open Liberty shared resources
+    COPY --from=infinispan-client /opt/ol/wlp/usr/shared/resources/infinispan /opt/ol/wlp/usr/shared/resources/infinispan
+    
+    # Instruct docker-server.sh to use Infinispan for session caching.
+    # This should be set to the Infinispan service name.
+    # TIP - Run the following oc/kubectl command with admin permissions to determine this value:
+    #       oc get infinispan -o jsonpath={.items[0].metadata.name}
+    ENV INFINISPAN_SERVICE_NAME=example-infinispan
+    
+    # Uncomment and set to override auto detected values.
+    # These are normally not needed, and are for advanced scenarios
+    # One such scenario would be the Infinispan deployment and Liberty deployment being in different namespaces/projects.
+    #ENV INFINISPAN_HOST=
+    #ENV INFINISPAN_PORT=
+    #ENV INFINISPAN_USER=
+    #ENV INFINISPAN_PASS=
+    
+    # This script will add the requested XML snippets and grow image to be fit-for-purpose
+    RUN configure.sh
+    USER 1001
+    ```
+
+    *  **Mount Infinispan Secret** - Finally, the Infinispan generated secret must be mounted as a volume under the mount point of `/config/liberty-infinispan-secret` on Liberty containers. When using the Infinispan Operator, this secret is automatically generated as part of the Infinispan service with the name of `<INFINISPAN_CLUSTER_NAME>-generated-secret`. See the `volumes` and `volumeMounts` portions of the YAML below for an example of mounting this secret.
```

Comment:
This is great integration with k8s.  My two comments:
(1) Folks will still first develop with just plain containers (i.e. `appsody run`), so there should be a way to run a container that connects to infinispan using just plain docker / podman.  I suggest you change the values of `infinispan-client-sessioncache.xml` to be variables, so that someone testing this could just either set them via environment variables (not suitable for production, but good for quick dev).

(2) Can you please open a feature in the [Open Liberty Operator](https://github.com/OpenLiberty/open-liberty-operator/issues) to have an option in the `OpenLibertyApplication` CRD that would automatically fetch the infinispan secret and mount in the container?

Reply:
I now see that related to (1), you are checking for those environment variables in `docker-server.sh`.  I still think the proposal in (1) would be better, because it is more clear to customers that these XML variables are configurable - as they won't know to look inside `docker-server.sh` for the logic that checks for it.

Reply:
The following issue has been opened.
https://github.com/OpenLiberty/open-liberty-operator/issues/135